src/WordPress/Blueprints/Model/BlueprintBuilder.php

```diff
@@ -123,19 +128,34 @@ public function withFiles( $files ) {
 	public function withFile( $path, $data ) {
 		return $this->addStep(
 			( new WriteFileStep() )
-				->setPath( 'wordpress.txt' )
+				->setPath( 'WordPress.txt' )
 				->setData( $data )
 		);
 	}
 
-	public function downloadWordPress( string|ResourceDefinitionInterface $wpZip = null ) {
-		$this->prependStep( ( new DownloadWordPressStep() )
+	public function remove( $path ) {
+		return $this->addStep(
+			( new RmStep() )
+			->setPath( $path )
+		);
+	}
+
+	public function makeDirectory( $path ) {
```

Comment:
Let's remove these methods for now.

First, this is additional code to maintain and keep in sync with the MkdirStep signature. 

Second, it doesn't expose all the MkdirStep features, e.g. you can't specify progress information here.

Third, and this is the most important one, the Builder API needs a deeper design dive to make sure the method names communicate the semantics well. The name `makeDirectory` doesn't make the consequences apparent. Does it add a new step? Does it create a directory immediately? If it's called together with some shorthand methods, what will be the order of execution? Also, if a Blueprint plugin registers more steps, there won't be shorthand methods for them – is that okay or not?

Reply:
I've reviewed the Blueprint Builder. 
It's a great idea, but still in its infancy. Many methods are implemented in a way that does not fully expose their features. Yet, using the Blueprint Builder for test data is a boon. 

If this Builder is still so floaty, maybe we can let those methods be and create a full issue to ideate on it and on all considered mapping strategies. 

I had many of the questions you asked when I was writing tests that coincidentally tested mapping to the `InstallPluginStep` and realized it has a shorthand. Blueprint Builder allows only for the shorthand. The JSON schema allows for both shorthand and steps. I have found this really confusing. First, you can't ideally reflect one mapping strategy to another i.e. there is no way to Build a Blueprint so that it is identical to one achieved via JSON. I don't exactly know how I feel about it, but I know it is not stellar.

See this test:
https://github.com/WordPress/blueprints/blob/7415d32f5886c2c6f3f4e8ea70f800e2874d6644/tests/JsonMapper/JsonMapperTest.php#L82-L111

It fails. The result, as one can expect, is that the mapper returns an object with some plugins in the shorthand plugin list, and some as steps. This discrepancy is then corrected in the `BlueprintCompiler` with [expandShorthandsIntoSteps](https://github.com/WordPress/blueprints/blob/f2ddfa54a97220249201535f1d37726c416f915d/src/WordPress/Blueprints/Compile/BlueprintCompiler.php#L32). Yet this feels late. Ideally all 4 mapping strategies: `from Blueprint Builder`, `from Blueprint`, `from JSON` and `from JSON string` should yield the same object.

Reply:
Does this need a separate issue or is it covered by the general scheme ponderings?

Ah. Important remark - I love the idea of shorthands, just this one is not sufficiently fleshed out yet.

Reply:
> If this Builder is still so floaty, maybe we can let those methods be and create a full issue to ideate on it and on all considered mapping strategies.

No, that will generate more noise to process later on. Let's just keep it as it is for now.

Reply:
Done.